app/Http/Controllers/GroupController.php

```diff
@@ -929,9 +929,16 @@ public function ajaxDeleteImage($group_id, $id, $path)
 
     public function getMakeHost($group_id, $user_id, Request $request)
     {
-        //Has current logged in user got permission to add host
-        if ((FixometerHelper::hasRole(Auth::user(), 'Host') && FixometerHelper::userIsHostOfGroup($group_id, Auth::id())) || FixometerHelper::hasRole(Auth::user(), 'Administrator')) {
-            $group = Group::find($group_id);
+        // Has current logged in user got permission to add host?
+        // - Is a host of the group.
+        // - Is a network coordinator of a network which the group is in.
+        // - Is an Administrator
+        $group = Group::find($group_id);
+        $loggedInUser = Auth::user();
+
+        if (($loggedInUser->hasRole('Host') && FixometerHelper::userIsHostOfGroup($group_id, $loggedInUser->id)) ||
```

Comment:
I've changed this to prefer methods that are on the user rather than in FixometerHelper or inline.  We would probably want to factor this out in to a policy in future..